### DISCUSSION TITLE
REST API available

### DISCUSSION THREAD

**Discussion by manalishi70** [thirtythreeforty/neolink#241] [Q&A]
I'm using RLC-410-5MP and Homeassistant together with AI for object and face detection. This camera type supports **RestfulAPI** to recieve PIR state. (http://xxx.xxx.xxx.xx/api.cgi?cmd=GetMdState&user=......). Using this api avoids the 6sec lag compared to the rtsp stream when it comes to motion detection. 
Is it possible to implement this in neolink or do the **lumus type cams** and other battery operated cams not support api at all?

**Comment by QuantumEntangledAndy** [comment]
I believe there is no api for a lot of these cameras we deal with. I don't even think port 80 is even open on my camera.

The whole idea of neolink was to gain access to these "dumb" cameras that lacked a lot of quality features.

**Reply by QuantumEntangledAndy** [reply]
In terms of your lag issue I should note that we do expose the motion detection state of the camera over mqtt (on one of my experimental branches). Buts it not done using the restapi, we instead login and communicate to the camera using their own bcprotocol.

One day I hope we can write an onvif layer ontop of neolink so that the motion state can be polled using standard techniques.
